solidity/contracts/bridge/Bridge.sol

```diff
@@ -440,49 +391,65 @@ contract Bridge is Ownable, EcdsaWalletOwner {
+    ///        UTXO of the given active wallet, as currently known on the
+    ///        Ethereum chain. If there is no active wallet at the moment, or
+    ///        the active wallet has no main UTXO, this parameter can be
+    ///        empty as it is ignored.
+    ///      - Wallet creation must not be in progress
+    ///      - If the active wallet is set, one of the following
+    ///        conditions must be true:
+    ///        - The active wallet BTC balance is above the minimum threshold
+    ///          and the active wallet is old enough, i.e. the creation period
+    ///           was elapsed since its creation time
+    ///        - The active wallet BTC balance is above the maximum threshold
+    function requestNewWallet(BitcoinTx.UTXO calldata activeWalletMainUtxo)
+        external
+    {
+        wallets.requestNewWallet(activeWalletMainUtxo);
     }
 
     /// @notice A callback function that is called by the ECDSA Wallet Registry
     ///         once a new ECDSA wallet is created.
-    /// @dev ECDSA Wallet Registry assigns a unique wallet ID generated by hashing
-    ///      wallet's public key with keccak256. In this contract we expect
-    ///      to identify wallets by Bitcoin's hash160, hence we map the wallets
-    ///      between those two.
     /// @param ecdsaWalletID Wallet's unique identifier.
-    /// @param publicKeyY Wallet's public key's X coordinate.
+    /// @param publicKeyX Wallet's public key's X coordinate.
     /// @param publicKeyY Wallet's public key's Y coordinate.
+    /// @dev Requirements:
+    ///      - The only caller authorized to call this function is `registry`
+    ///      - Given wallet data must not belong to an already registered wallet
     function __ecdsaWalletCreatedCallback(
         bytes32 ecdsaWalletID,
         bytes32 publicKeyX,
         bytes32 publicKeyY
     ) external override {
-        require(
-            msg.sender == address(ecdsaWalletRegistry),
-            "Caller is not the ECDSA Wallet Registry"
-        );
-
-        // Compress wallet's public key and calculate Bitcoin's hash160 of it.
-        bytes20 walletPubKeyHash = bytes20(
-            EcdsaLib.compressPublicKey(publicKeyX, publicKeyY).hash160()
-        );
-
-        require(
-            wallets[walletPubKeyHash].state == WalletState.Unknown,
-            "ECDSA wallet has been already registered"
-        );
+        wallets.registerNewWallet(ecdsaWalletID, publicKeyX, publicKeyY);
+    }
 
-        wallets[walletPubKeyHash].ecdsaWalletID = ecdsaWalletID;
-        wallets[walletPubKeyHash].state = WalletState.Active;
+    /// @notice Gets details about a registered wallet.
+    /// @param walletPubKeyHash The 20-byte wallet public key hash (computed
+    ///        using Bitcoin HASH160 over the compressed ECDSA public key)
+    /// @return Wallet details.
+    function getRegisteredWallet(bytes20 walletPubKeyHash)
```

Comment:
I would consider renaming it to `getWallet`. There are no other wallets known to the `Bridge` than these registered. Same about `setWallet` vs `setRegisteredWallet` in the stub.

Reply:
See https://github.com/keep-network/tbtc-v2/pull/172/commits/b326f105c59c3dc53685704b66ba9e4b8b0e1be8